interfaces/apparmor/backend.go

```diff
@@ -329,6 +330,19 @@ func (b *Backend) Setup(snapInfo *snap.Info, opts interfaces.ConfinementOptions,
 	if snapName == "core" && !release.OnClassic {
 		if li, err := filepath.Glob(filepath.Join(dirs.SystemApparmorCacheDir, "*")); err == nil {
 			for _, p := range li {
+				// On systems with unified cache directory,
+				// don't remove the snap profiles, only system
+				// and snap-confine profiles. snap-confine
+				// profiles are like the following:
+				// - usr.lib.snapd.snap-confine.real
+				// - usr.lib.snapd.snap-confine
+				// - snap.core.NNNN.usr.lib.snapd.snap-confine
+				// - snap-confine.core.NNNN
+				// TODO: also the "snapd" snap here soon
```

Comment:
we now have "snapd" snap based systems,  it might well be time to implement the TODOs and not leave them as TODO. let's see what @mvo thinks

Reply:
I agree with pedronis

Reply:
> we now have "snapd" snap based systems, it might well be time to implement the TODOs and not leave them as TODO. let's see what @mvo thinks

Please note that it is very important that this makes 2.38 since I can't update AppArmor in the archive much later than that due to distro timelines. Since this TODO was already there and unrelated to this PR, I suggest it be done in a separate PR.